als-server/shared/src/main/scala/org/mulesoft/als/server/workspace/FutureLock.scala

```diff
@@ -0,0 +1,65 @@
+package org.mulesoft.als.server.workspace
+
+import java.util.concurrent.atomic.AtomicBoolean
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Future, Promise}
+
+class FutureLock() {
```

Comment:
no nos sirven los `Semaphore`s que ya existen?

Reply:
Semaphore de Java bloquea, en JS no anda

Reply:
https://github.com/scala-js/scala-js/issues/4102

entenderia que deberia estar implementado, seguro que no funca? (tambien mencionan una alternativa anterior que es usando un `AtomicInteger`)

Reply:
No funciona, class not found cuando buildea

Reply:
Por la fecha del commit/merge, parece estar incluido recien en el release v1.5.0 https://github.com/scala-js/scala-js/releases/tag/v1.5.0